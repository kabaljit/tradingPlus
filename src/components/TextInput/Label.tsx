import React from 'react';
import { Animated, LayoutChangeEvent } from 'react-native';

import { scale } from '../../utils/layout';
import theme from '../../theme';

import { StyledLabel } from './TextInput.styles';

interface ILabelProps {
  label: string;
  animation: any;
  state: 'error' | 'default' | 'focused' | 'complete';
}

const Label = ({ label, animation, state }: ILabelProps) => {
  const [layout, setLayout] = React.useState({ width: 0, height: 0 });

  const labelScale = 0.75; // desiired font-size / font-size
  const labelWidth = layout.width;
  const labelHeight = layout.height;
  const labelHalfWidth = labelWidth / 2;
  const labelHalfHeight = labelHeight / 2;

  // This is the equivilant of transform-origin in the web. Proceed with caution.
  const baseLabelTranslateX =
    -1 * (labelHalfWidth - (labelScale * labelWidth) / 2) +
    (1 - labelScale) * scale(18);

  const baseLabelTranslateY = -labelHalfHeight - 10;

  const containerStyle = {
    transform: [
      {
        translateY: animation.interpolate({
          inputRange: [0, 1],
          outputRange: [0, baseLabelTranslateY],
        }),
      },
      {
        translateX: animation.interpolate({
          inputRange: [0, 1],
          outputRange: [0, baseLabelTranslateX],
        }),
      },
      {
        scale: animation.interpolate({
          inputRange: [0, 1],
          outputRange: [1, labelScale],
        }),
      },
    ],
  };

  const labelStyles = {
    color: animation.interpolate({
      inputRange: [0, 1],

      // TODO: UPDATE THE COLOR FOR EACH STATE input.label[state]],
      outputRange: [theme.colors.tradingZ.white, theme.colors.tradingZ.success],
    }),
  };

  return (
    <Animated.View
      onLayout={({ nativeEvent: { layout: newLayout } }: LayoutChangeEvent) =>
        setLayout(newLayout)
      }
      style={containerStyle}
    >
      <StyledLabel style={labelStyles}>{label}</StyledLabel>
    </Animated.View>
  );
};

export default Label;
