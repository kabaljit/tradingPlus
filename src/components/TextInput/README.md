# Form Element - Text Input

[![View in Playground](https://img.shields.io/badge/%20View%20in-⛰️%20Playground-262834.svg?style=flat-square)](playground/screens/FormElements.tsx)

`TextInput` components do not handle their own validation, they rely on the `InputWrapper` to add validation.

The `TextInput` has been designed to be easily composable, i.e we should be able to extend this component to build others and not create it again and again from scratch.

### Usage

**Standard**

```jsx
<TextInput label="This is a label" />
```

**With validation**

```jsx
<InputWrapper error={true} errorMessage="This is an error message">
  <Dropdown
    value="2"
    error={true}
    items={[
      { label: 'Options 1 ', value: '1' },
      { label: 'Options 2 ', value: '2' },
      { label: 'Options 3', value: '3' },
    ]}
    onValueChange={() => {}}
  />
</InputWrapper>
```

