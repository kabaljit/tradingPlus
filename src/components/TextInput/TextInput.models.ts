import { TextInputProps as RNTextInputProps } from 'react-native';
import { ReactElement } from 'react';

import theme from '../../theme';

export interface TextInputProps extends RNTextInputProps {
  accessoryLeft?: ReactElement<any>;
  accessoryRight?: ReactElement<any>;
  label?: string;
  error?: boolean;
  formatText?: (text: string) => string;
  dangerouslySetStyles?: IdangerouslySetFontStyles & IdangerouslySetInputStyles;
  align?: 'left' | 'right';
  backgroundColor?: string;
}

export interface IdangerouslySetInputStyles {
  input?: {
    minHeight?: number;
    paddingTop?: number;
    lineHeight?: number;
  };
}

export interface IdangerouslySetFontStyles {
  font?: {
    size: number;
    fontWeight?: keyof typeof theme.fontWeights;
  };
}

export enum InputStates {
  DEFAULT = 'default',
  FOCUSED = 'focused',
  ERROR = 'error',
  COMPLETE = 'complete',
}
