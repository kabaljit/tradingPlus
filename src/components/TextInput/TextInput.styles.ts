import styled from 'styled-components/native';
import { Animated } from 'react-native';

import theme from '../../theme';
import { paragraphSizes } from '../Typography/mappings';
import { scale } from '../../utils/layout';

import {
  IdangerouslySetFontStyles,
  IdangerouslySetInputStyles,
} from './TextInput.models';

// TODO: When refactoring theme to include contextual sizes,
// move these in there.
export const height = scale(59);
export const fontSize = paragraphSizes.medium;
export const labelFontSize = paragraphSizes.small;
export const helperFontSize = paragraphSizes.extraSmall;
export const helperPaddingVertical = theme.spacing[1];
export const helperPaddingHorizontal = theme.spacing[1];
export const iconSize = height * 0.75;

const borderColorForState = (isError: boolean, isActive: boolean) => {
  if (isError) return theme.colors.tradingZ.error;
  if (isActive) return theme.colors.tradingZ.darkSilver;
  return theme.colors.tradingZ.charcoal;
};

export const FakeInputBorder = styled.View<{
  active?: boolean;
  error?: boolean;
}>`
  position: absolute;
  bottom: ${(props) => (props.active ? 0 : 0.5)}px;
  left: 0;
  right: 0;
  background-color: ${(p) => borderColorForState(!!p.error, !!p.active)};
  height: ${(props) => (props.active ? 2 : 1)}px;
`;

// New

export const AnimatedBackground = styled(Animated.View)<{ minHeight?: number }>`
  position: relative;
  background-color: ${theme.colors.tradingZ.charcoal};
  min-height: ${(p) => p.minHeight || scale(56)}px;
`;

export const InnerRow = styled.View`
  flex-direction: row;
  flex: 1;
  align-items: center;
`;

export const AnimatedBorder = styled(Animated.View)`
  width: 100%;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  height: ${scale(1)}px;
`;

export const InputInner = styled.View`
  position: relative;
  flex: 1;
`;

export const Input = styled.TextInput<{
  textOnly?: boolean;
  dangerouslySetStyles?: IdangerouslySetFontStyles & IdangerouslySetInputStyles;
  align: 'left' | 'right';
  backgroundColor: string;
}>`
  min-height: ${(p) => p.dangerouslySetStyles?.input?.minHeight || scale(56)}px;
  font-size: ${(p) => p.dangerouslySetStyles?.font?.size || scale(16)}px;
  color: ${theme.colors.tradingZ.white};
  background-color: ${(p) => p.backgroundColor ?? 'transparent'};
  padding-left: ${scale(17)}px;
  line-height: ${(p) =>
    p.dangerouslySetStyles?.input?.lineHeight ||
    (p.dangerouslySetStyles?.font && p.dangerouslySetStyles.font.size * 1.18) ||
    scale(19)}px;
  text-align: ${(p) => p.align};
  font-weight: ${(p) => p.dangerouslySetStyles?.font?.fontWeight || 'normal'}
    ${(p) =>
      p.dangerouslySetStyles?.input?.paddingTop &&
      `padding-top: ${p.dangerouslySetStyles?.input?.paddingTop}`}px;
  ${(p) => !p.textOnly && `padding-top: ${scale(12)}px;`}
  ${(p) =>
    p.multiline &&
    `
    padding-top: ${scale(23)}px;
    padding-bottom: ${scale(10)}px
  `};
`;

export const StyledLabel = styled(Animated.Text)`
  position: absolute;
  /* (HEIGHT / 2) - (FONT_SIZE / 2) */
  top: ${scale(20)}px;
  font-size: ${scale(16)}px;
  line-height: ${scale(16)}px;
  left: ${scale(16)};
`;

export const RightAccessoryWrapper = styled.View`
  padding-right: ${scale(16)}px;
`;

export const LeftAccessoryWrapper = styled.View`
  padding-left: ${scale(16)}px;
`;
