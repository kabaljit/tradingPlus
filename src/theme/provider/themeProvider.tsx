import { createTheming } from "@callstack/react-theme-provider";
import React from "react";

import { tradeTheme } from "./mytheme";
import { ITheme, IThemeProvider } from "./models";

const {
  ThemeProvider: BareThemeProvider,
  withTheme,
  useTheme,
} = createTheming<ITheme>(tradeTheme);

export const ThemeProvider: React.FunctionComponent<IThemeProvider> = ({
  children,
  theme,
}) => <BareThemeProvider theme={theme}>{children}</BareThemeProvider>;

export { useTheme, withTheme };
