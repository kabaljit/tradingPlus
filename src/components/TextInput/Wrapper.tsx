import React from 'react';
import { Animated } from 'react-native';

import {
  AnimatedBackground,
  AnimatedBorder,
  InnerRow,
} from './TextInput.styles';
import { IdangerouslySetInputStyles } from './TextInput.models';
import theme from '../../theme';

interface IWrapperProps {
  animation: Animated.AnimatedValue;
  state: 'default' | 'focused' | 'error' | 'complete';
  dangerouslySetStyles?: IdangerouslySetInputStyles;
}

const Wrapper: React.FunctionComponent<IWrapperProps> = ({
  children,
  animation,
  state,
  dangerouslySetStyles,
}) => {
  const animatedStyles = {
    backgroundColor: animation.interpolate({
      inputRange: [0, 1],
      outputRange: [
        theme.colors.tradingZ.transparent,
        theme.colors.tradingZ.charcoal,
      ],
    }),
  };

  const animatedBorderStyles = {
    backgroundColor: animation.interpolate({
      inputRange: [0, 1],
      outputRange: [
        theme.colors.tradingZ.lightGrey,
        theme.colors.tradingZ.white,
      ],
    }),
  };

  return (
    <AnimatedBackground
      minHeight={dangerouslySetStyles?.input?.minHeight}
      style={animatedStyles}
    >
      <InnerRow>{children}</InnerRow>
      <AnimatedBorder style={animatedBorderStyles} />
    </AnimatedBackground>
  );
};

export default Wrapper;
