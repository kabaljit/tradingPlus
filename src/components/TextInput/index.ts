import { withTheme } from '../../theme/provider/themeProvider';

import { TextInput } from './TextInput';
import type { TextInputProps as TextInputPropsBare } from './TextInput.models';

export {
  FakeInputBorder,
  height as defaultTextInputHeight,
  fontSize as defaultTextInputFontSize,
  labelFontSize as defaultTextInputLabelFontSize,
  helperFontSize as defaultTextInputHelperFontSize,
  helperPaddingVertical as defaultTextInputHelperPaddingVertical,
  helperPaddingHorizontal as defaultTextInputHelperPaddingHorizontal,
  iconSize as defaultTextInputIconSize,
} from './TextInput.styles';

export type TextInputProps = Omit<TextInputPropsBare, 'theme'>;
export default withTheme(TextInput);
