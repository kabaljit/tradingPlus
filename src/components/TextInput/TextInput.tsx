/* eslint-disable @typescript-eslint/ban-ts-ignore */
import * as React from 'react';
import {
  View,
  Animated,
  Easing,
  TextInput as RNTextInput,
  TextInputFocusEventData,
  NativeSyntheticEvent,
} from 'react-native';

import { TextInputProps, InputStates } from './TextInput.models';
import { Input, InputInner } from './TextInput.styles';
import Label from './Label';
import Wrapper from './Wrapper';
import theme from '../../theme';

export const TextInput = React.forwardRef<RNTextInput, TextInputProps>(
  (props, ref) => {
    const {
      align = 'left',
      value,
      error,
      placeholder,
      dangerouslySetStyles,
      label,
      formatText,
      onChangeText,
      onFocus,
      onBlur,
    } = props;

    const innerRef = ref ?? React.useRef();

    const labelAnimation = React.useMemo(() => new Animated.Value(0), []);
    const baseAnimation = React.useMemo(() => new Animated.Value(0), []);
    const [inputState, setInputState] = React.useState<InputStates>(
      InputStates.DEFAULT
    );

    const triggerErrorState = () => {
      setInputState(InputStates.ERROR);
      labelAnimation.setValue(1);
    };

    React.useEffect(() => {
      if (error) {
        triggerErrorState();
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [error]);

    // Helper for handling animations in a consistent fashion
    const startAnimation = (animation: any, options: any, callback?: any) => {
      const formattedOptions = {
        duration: 250,
        easing: Easing.ease,
        ...options,
      };
      Animated.timing(animation, formattedOptions).start(callback);
    };

    const startLabelAnimation = () => {
      const options = {
        toValue: inputState === InputStates.FOCUSED ? 0 : 1,
        useNativeDriver: false,
      };
      startAnimation(labelAnimation, options);
    };

    const startBaseAnimation = () => {
      const options = {
        toValue: inputState === InputStates.FOCUSED ? 0 : 1,
        // This is important as we're using this to animate colours, you
        // can't animate colours using the native driver
        useNativeDriver: false,
      };
      startAnimation(baseAnimation, options);
    };

    React.useEffect(() => {
      if (value?.length && !error) {
        setInputState(InputStates.COMPLETE);
        labelAnimation.setValue(1);
      } else if (placeholder) {
        setInputState(InputStates.DEFAULT);
        labelAnimation.setValue(1);
      }
    }, [labelAnimation, placeholder, value]);

    const handleFocus = (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setInputState(InputStates.FOCUSED);
      startBaseAnimation();
      !value && startLabelAnimation();
      onFocus?.(e);
    };

    const handleBlur = (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      if (props.error) {
        triggerErrorState();
      } else {
        setInputState(
          value?.length ? InputStates.COMPLETE : InputStates.DEFAULT
        );
        !value && !props.placeholder && startLabelAnimation();
      }
      startBaseAnimation();
      onBlur?.(e);
    };

    const handleChangeText = React.useCallback(
      (text: string) => {
        if (formatText) {
          text = formatText(text);
        }

        // @ts-ignore
        innerRef?.current?.setNativeProps({ text });

        onChangeText && onChangeText(text);
      },
      [formatText, onChangeText, innerRef]
    );

    const formattedValue = React.useMemo(
      () => (formatText && value ? formatText(value) : value),
      [value, formatText]
    );

    return (
      <Wrapper
        dangerouslySetStyles={dangerouslySetStyles}
        animation={baseAnimation}
        state={inputState}
      >
        {!!props.accessoryLeft && (
          <View testID="accessory-left">
            {React.cloneElement(props.accessoryLeft, {
              state: inputState,
              animation: baseAnimation,
            })}
          </View>
        )}
        <InputInner>
          {label && (
            <Label
              label={label}
              animation={labelAnimation}
              state={inputState}
            />
          )}
          <Input
            textOnly={!label}
            // @ts-ignore
            ref={innerRef}
            value={formattedValue}
            dangerouslySetStyles={dangerouslySetStyles}
            align={align}
            {
              ...props
              /* Below props need to be hooked to our own definitions for
              TextInput to work properly. If you also want to pass, let's say,
              a custom onChangeText, call it from within the method that handles
              our own logic. */
            }
            onFocus={handleFocus}
            onBlur={handleBlur}
            onChangeText={handleChangeText}
            selectionColor={theme.colors.tradingZ.blue}
            textAlignVertical="bottom"
            placeholderTextColor="#C0C0C5"
          />
        </InputInner>
        {!!props.accessoryRight && (
          <View testID="accessory-right">
            {React.cloneElement(props.accessoryRight, {
              state: inputState,
              animation: baseAnimation,
            })}
          </View>
        )}
      </Wrapper>
    );
  }
);
