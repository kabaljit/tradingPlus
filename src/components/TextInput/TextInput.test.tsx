import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '../../theme/provider/themeProvider';

import TextInput from '.';

describe('Given a TextInput', () => {
  describe('when the component is rendered', () => {
    it('should match the snapshot', () => {
      const tree = renderer
        .create(
          <ThemeProvider>
            <TextInput value="this is a test" />
          </ThemeProvider>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Given a multi-line TextInput', () => {
  describe('when the component is rendered', () => {
    it('should match the snapshot', () => {
      const tree = renderer
        .create(
          <ThemeProvider>
            <TextInput multiline={true} value="this is a test" />
          </ThemeProvider>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
