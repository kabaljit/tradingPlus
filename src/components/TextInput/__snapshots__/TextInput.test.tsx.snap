// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a TextInput when the component is rendered should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(247, 248, 252, 1)",
      "minHeight": 74.4,
      "position": "relative",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "position": "relative",
          },
        ]
      }
    >
      <TextInput
        align="left"
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        selectionColor="#FF4D54"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontSize": 21.25714285714286,
              "fontWeight": "normal",
              "lineHeight": 25.242857142857144,
              "minHeight": 74.4,
              "paddingLeft": 22.58571428571429,
              "textAlign": "left",
            },
          ]
        }
        textAlignVertical="bottom"
        textOnly={true}
        theme={
          Object {
            "colorPalette": Object {
              "green": Array [
                "#0BA866",
              ],
              "grey": Array [
                "#F7F8FC",
                "#EDEDF4",
                "#E1E1EA",
                "#D4D5DD",
                "#BCBCC9",
                "#9699AA",
                "#7C7E8D",
                "#676978",
                "#484A58",
              ],
              "mango": Array [
                "#FFB900",
                "#FFD566",
              ],
              "midnightBlue": Array [
                "#262835",
              ],
              "red": Array [
                "#FFFEF3",
                "#FFE0D7",
                "#FFC2BB",
                "#FFA5A0",
                "#FF8886",
                "#FF6B6C",
                "#FF4D54",
                "#FF333C",
                "#FF1A23 ",
              ],
              "seaGreen": Array [
                "#D5F1F3",
                "#C1EAED",
                "#ACE3E7",
                "#9CDEE1",
                "#82D5DA",
                "#6DCDD3",
                "#59C7CE",
                "#2FB9C2",
                "#238B92",
                "#185D61",
              ],
              "skyBlue": Array [
                "#F7FAFD",
                "#E5F2FC",
                "#C0DEF8",
                "#99CAF4",
                "#75B6EF",
                "#5DA6EC",
                "#4C97E8",
                "#4689DA",
                "#3871C2",
                "#3260B0",
              ],
              "white": Array [
                "#FFFFFF",
              ],
              "yellow": Array [
                "#FFF8E4",
              ],
            },
            "context": Object {
              "colors": Object {
                "actions": Object {
                  "background": Object {
                    "link": Object {
                      "complete": "",
                      "default": "transparent",
                      "disabled": "transparent",
                      "error": "",
                      "focused": "transparent",
                      "loading": "transparent",
                    },
                    "primary": Object {
                      "complete": "",
                      "default": "#FF4D54",
                      "disabled": "#E1E1EA",
                      "error": "",
                      "focused": "",
                      "loading": "#E1E1EA",
                    },
                    "secondary": Object {
                      "complete": "",
                      "default": "transparent",
                      "disabled": "transparent",
                      "error": "",
                      "focused": "transparent",
                      "loading": "transparent",
                    },
                  },
                  "foreground": Object {
                    "link": Object {
                      "complete": "",
                      "default": "#FFFEF3",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                    "primary": Object {
                      "complete": "",
                      "default": "#FFFFFF",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                    "secondary": Object {
                      "complete": "",
                      "default": "#FFFEF3",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                  },
                },
                "input": Object {
                  "background": Object {
                    "complete": "#F7FAFD",
                    "default": "#F7F8FC",
                    "error": "#FFFEF3",
                    "focused": "#FFFFFF",
                  },
                  "border": Object {
                    "complete": "#4689DA",
                    "default": "#9699AA",
                    "error": "#FFE0D7",
                    "focused": "#FF4D54",
                  },
                  "icon": Object {
                    "complete": "#262835",
                    "default": "#262835",
                    "error": "#262835",
                    "focused": "#262835",
                  },
                  "label": Object {
                    "complete": "#9699AA",
                    "default": "#9699AA",
                    "error": "#FFE0D7",
                    "focused": "#9699AA",
                  },
                  "outerLabel": Object {
                    "complete": "#262835",
                    "default": "#262835",
                    "error": "#262835",
                    "focused": "#262835",
                  },
                },
                "states": Object {
                  "error": Object {
                    "complete": "",
                    "default": "#FFFEF3",
                    "error": "",
                    "focused": "",
                  },
                  "loading": Object {
                    "complete": "",
                    "default": "#FFB900",
                    "error": "",
                    "focused": "",
                  },
                },
                "typography": Object {
                  "blue": "#3260B0",
                  "error": "#FF8886",
                  "inverse": "#D4D5DD",
                  "link": "#3871C2",
                  "mango": "#FFB900",
                  "primary": "#262835",
                  "secondary": "#F7F8FC",
                  "success": "#0BA866",
                  "tertiary": "#7C7E8D",
                  "white": "#FFFFFF",
                },
              },
              "fontSizes": Object {
                "paragraph": Object {
                  "large": 31.5,
                  "normal": 27.5,
                  "small": 23.5,
                },
                "title": Object {
                  "large": 63,
                  "normal": 47.5,
                  "small": 35.5,
                },
              },
              "fontWeights": Object {
                "paragraph": Object {
                  "bold": "700",
                  "medium": "600",
                  "regular": "400",
                },
                "title": Object {
                  "bold": "700",
                  "medium": "600",
                  "regular": "400",
                },
              },
              "sizes": Object {
                "buttonHeight": 108.5,
                "formElementHeight": 91,
                "formElementIconHeight": 67,
                "hairline": 0.5,
                "listItemHeight": 108.5,
              },
              "spacing": Object {
                "element": 16,
                "screenEdge": 31.5,
              },
            },
          }
        }
        underlineColorAndroid="transparent"
        value="this is a test"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(150, 153, 170, 1)",
        "bottom": 0,
        "height": 2.6571428571428575,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`Given a multi-line TextInput when the component is rendered should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(247, 250, 253, 1)",
      "minHeight": 74.4,
      "position": "relative",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "position": "relative",
          },
        ]
      }
    >
      <TextInput
        align="left"
        allowFontScaling={true}
        multiline={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        selectionColor="#FF4D54"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontSize": 21.25714285714286,
              "fontWeight": "normal",
              "lineHeight": 25.242857142857144,
              "minHeight": 74.4,
              "paddingBottom": 13.285714285714286,
              "paddingLeft": 22.58571428571429,
              "paddingTop": 30.557142857142857,
              "textAlign": "left",
            },
          ]
        }
        textAlignVertical="bottom"
        textOnly={true}
        theme={
          Object {
            "colorPalette": Object {
              "green": Array [
                "#0BA866",
              ],
              "grey": Array [
                "#F7F8FC",
                "#EDEDF4",
                "#E1E1EA",
                "#D4D5DD",
                "#BCBCC9",
                "#9699AA",
                "#7C7E8D",
                "#676978",
                "#484A58",
              ],
              "mango": Array [
                "#FFB900",
                "#FFD566",
              ],
              "midnightBlue": Array [
                "#262835",
              ],
              "red": Array [
                "#FFFEF3",
                "#FFE0D7",
                "#FFC2BB",
                "#FFA5A0",
                "#FF8886",
                "#FF6B6C",
                "#FF4D54",
                "#FF333C",
                "#FF1A23 ",
              ],
              "seaGreen": Array [
                "#D5F1F3",
                "#C1EAED",
                "#ACE3E7",
                "#9CDEE1",
                "#82D5DA",
                "#6DCDD3",
                "#59C7CE",
                "#2FB9C2",
                "#238B92",
                "#185D61",
              ],
              "skyBlue": Array [
                "#F7FAFD",
                "#E5F2FC",
                "#C0DEF8",
                "#99CAF4",
                "#75B6EF",
                "#5DA6EC",
                "#4C97E8",
                "#4689DA",
                "#3871C2",
                "#3260B0",
              ],
              "white": Array [
                "#FFFFFF",
              ],
              "yellow": Array [
                "#FFF8E4",
              ],
            },
            "context": Object {
              "colors": Object {
                "actions": Object {
                  "background": Object {
                    "link": Object {
                      "complete": "",
                      "default": "transparent",
                      "disabled": "transparent",
                      "error": "",
                      "focused": "transparent",
                      "loading": "transparent",
                    },
                    "primary": Object {
                      "complete": "",
                      "default": "#FF4D54",
                      "disabled": "#E1E1EA",
                      "error": "",
                      "focused": "",
                      "loading": "#E1E1EA",
                    },
                    "secondary": Object {
                      "complete": "",
                      "default": "transparent",
                      "disabled": "transparent",
                      "error": "",
                      "focused": "transparent",
                      "loading": "transparent",
                    },
                  },
                  "foreground": Object {
                    "link": Object {
                      "complete": "",
                      "default": "#FFFEF3",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                    "primary": Object {
                      "complete": "",
                      "default": "#FFFFFF",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                    "secondary": Object {
                      "complete": "",
                      "default": "#FFFEF3",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                  },
                },
                "input": Object {
                  "background": Object {
                    "complete": "#F7FAFD",
                    "default": "#F7F8FC",
                    "error": "#FFFEF3",
                    "focused": "#FFFFFF",
                  },
                  "border": Object {
                    "complete": "#4689DA",
                    "default": "#9699AA",
                    "error": "#FFE0D7",
                    "focused": "#FF4D54",
                  },
                  "icon": Object {
                    "complete": "#262835",
                    "default": "#262835",
                    "error": "#262835",
                    "focused": "#262835",
                  },
                  "label": Object {
                    "complete": "#9699AA",
                    "default": "#9699AA",
                    "error": "#FFE0D7",
                    "focused": "#9699AA",
                  },
                  "outerLabel": Object {
                    "complete": "#262835",
                    "default": "#262835",
                    "error": "#262835",
                    "focused": "#262835",
                  },
                },
                "states": Object {
                  "error": Object {
                    "complete": "",
                    "default": "#FFFEF3",
                    "error": "",
                    "focused": "",
                  },
                  "loading": Object {
                    "complete": "",
                    "default": "#FFB900",
                    "error": "",
                    "focused": "",
                  },
                },
                "typography": Object {
                  "blue": "#3260B0",
                  "error": "#FF8886",
                  "inverse": "#D4D5DD",
                  "link": "#3871C2",
                  "mango": "#FFB900",
                  "primary": "#262835",
                  "secondary": "#F7F8FC",
                  "success": "#0BA866",
                  "tertiary": "#7C7E8D",
                  "white": "#FFFFFF",
                },
              },
              "fontSizes": Object {
                "paragraph": Object {
                  "large": 31.5,
                  "normal": 27.5,
                  "small": 23.5,
                },
                "title": Object {
                  "large": 63,
                  "normal": 47.5,
                  "small": 35.5,
                },
              },
              "fontWeights": Object {
                "paragraph": Object {
                  "bold": "700",
                  "medium": "600",
                  "regular": "400",
                },
                "title": Object {
                  "bold": "700",
                  "medium": "600",
                  "regular": "400",
                },
              },
              "sizes": Object {
                "buttonHeight": 108.5,
                "formElementHeight": 91,
                "formElementIconHeight": 67,
                "hairline": 0.5,
                "listItemHeight": 108.5,
              },
              "spacing": Object {
                "element": 16,
                "screenEdge": 31.5,
              },
            },
          }
        }
        underlineColorAndroid="transparent"
        value="this is a test"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(70, 137, 218, 1)",
        "bottom": 0,
        "height": 2.6571428571428575,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "width": "100%",
      }
    }
  />
</View>
`;
